### PR TITLE
Refactor KanikoBuild into KanikoArtifact and Cluster

### PR DESCRIPTION
**Pull request by priyawadhwa**
This PR refactors kaniko, taking the preexisting [KanikoBuild](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/schema/latest/config.go#L240) type and splitting it up into a per artifact [KanikoArtifact](https://github.com/priyawadhwa/skaffold-1/blob/kanikoartifact/pkg/skaffold/schema/latest/config.go#L555) and [Cluster](https://github.com/priyawadhwa/skaffold-1/blob/kanikoartifact/pkg/skaffold/schema/latest/config.go#L238) type. 

`KanikoArtifact` now contains artifact specified fields (buildcontext, cache, additional flags etc.), while`ClusterDetails` contains fields specific to the cluster (namespace, timeout, secrets). This makes more sense to me, since a buildcontext should be different for different artifacts (different bucket to pull from, etc). 

This PR updates the config to `v1beta8` as it represents a breaking change to the skaffold config. 

I also added an integration test to make sure kaniko can build multiple artifacts.

This refactor will ultimately make it much easier to implement #1549

### CODE REVIEW

File: pkg/skaffold/build/kaniko/run.go [hunk 1]
```diff
@@ -49,29 +46,26 @@ func (b *Builder) run(ctx context.Context, out io.Writer, artifact *latest.Artif
...
-		"--dockerfile", artifact.DockerArtifact.DockerfilePath,
+		"--dockerfile", kanikoArtifact.DockerfilePath,
 		"--context", context,
 		"--destination", tag,
 		"-v", logLevel().String()}
-	args = append(args, b.AdditionalFlags...)
-	args = append(args, docker.GetBuildArgs(artifact.DockerArtifact)...)
+
+	// TODO: remove since AdditionalFlags will be deprecated (priyawadhwa@)
+	args = append(args, kanikoArtifact.AdditionalFlags...)
```
Comment (balopat) [inline pkg/skaffold/build/kaniko/run.go]: let's put out a deprecation warning here into the log if `AdditionalFlags` are not empty?

File: pkg/skaffold/schema/defaults/defaults.go [hunk 2]
```diff
@@ -47,14 +47,21 @@ func Set(c *latest.SkaffoldPipeline) error {
...
+			setDefaultKanikoArtifactImage(a)
+			setDefaultKanikoArtifactBuildContext(a)
+			setDefaultKanikoDockerfilePath(a)
+		}
+	}
+
+	if err := withClusterConfig(c,
+		setDefaultClusterNamespace,
+		setDefaultClusterTimeout,
+		setDefaultClusterSecret,
```
Comment (balopat) [inline pkg/skaffold/schema/defaults/defaults.go]: How about 
```suggestion
		setDefaultClusterRegistrySecret,
```?

**Comment by priyawadhwa** [comment]
Ran into #1767 [here](https://source.cloud.google.com/results/invocations/2eab1328-6d15-43b0-b5a2-541e5ca0efbc/log), rerunning kokoro

**Comment by dgageot** [comment]
@priyawadhwa you might want to rebase on top of master instead, you'll get improvements to the integration tests

**Comment by priyawadhwa** [comment]
Thanks for the tip @dgageot !

**Pull request merged**